### PR TITLE
Feature/obs tests

### PR DESCRIPTION
**Pull request by ADCollard**
This PR fixes the EVA yaml for radiances, adds flexibility to [ush/ufoeval/run_ufo_hofx_test.sh](https://github.com/NOAA-EMC/GDASApp/blob/feature/obs_tests/ush/ufoeval/run_ufo_hofx_test.sh) to run a reduced EVA suite or none at all; and introduces the [test_yamls.sh](https://github.com/NOAA-EMC/GDASApp/blob/feature/obs_tests/ush/ufoeval/test_yamls.sh) script which loops through the YAMLs in [parm/atm/obs/testing](https://github.com/NOAA-EMC/GDASApp/tree/feature/obs_tests/parm/atm/obs/testing) and reports if they pass.

### CODE REVIEW

**Comment by ADCollard** [comment]
Example output:

```
(gdasapp) Orion-login-2[151] acollard$ ./test_yamls.sh
aircraft.yaml Fails (boo!)
amsua_n19.yaml Fails (boo!)
satwind.yaml Fails (boo!)
sfcship.yaml Fails (boo!)
sfc.yaml Fails (boo!)
sondes_ps.yaml Fails (boo!)
sondes.yaml Fails (boo!)
ssmis_f17.yaml Fails (boo!)
ssmis_f18.yaml Fails (boo!)
```

**Review by guillaumevernieres — DISMISSED** [review]
:+1: for the `boo` \ `yay` output, which is admittedly the only thing I have reviewed in this PR. 
Are you planning to add the tests to the `ci` eventually?

**Comment by ADCollard** [comment]
> 👍 for the `boo` \ `yay` output, which is admittedly the only thing I have reviewed in this PR. Are you planning to add the tests to the `ci` eventually?

I believe @CoryMartin-NOAA is keep to incorporate this into the CI.

File: ush/ufoeval/run_ufo_hofx_test.sh [hunk 1]
```diff
@@ -9,18 +9,55 @@
...
 #-------------------------------------------------------------
-#--------------- User modified options below -----------------
-obtype=$1
-cycle=${2:-2021080100}
+
+# ==============================================================================
+usage() {
+  set +x
+  echo
+  echo "Usage: $0 [-c cycle] [-x] [-s] [-h] instrument [cycle]"
```
Comment (CoryMartin-NOAA) [inline ush/ufoeval/run_ufo_hofx_test.sh]: Is this usage statement correct? Why `[cycle]` at the end?

File: ush/ufoeval/run_ufo_hofx_test.sh [hunk 2]
```diff
@@ -9,18 +9,55 @@
...
+      eva_stats_only=YES
+      ;;
+    h|\?|:)
+      usage
+      ;;
+  esac
+done
+shift $((OPTIND - 1))
 
 if [ $# -lt 1 ] || [ $# -gt 2 ]; then
```
Comment (CoryMartin-NOAA) [inline ush/ufoeval/run_ufo_hofx_test.sh]: can you not do instrument, cycle, and an eva option? that would be 3?

File: ush/ufoeval/run_ufo_hofx_test.sh [hunk 3]
```diff
@@ -9,18 +9,55 @@
...
-   exit 1
+   usage 
 fi
 
-machine=hera
+obtype=$1
+
+#--------------- User modified options below -----------------
+
+machine=orion
```
Comment (CoryMartin-NOAA) [inline ush/ufoeval/run_ufo_hofx_test.sh]: for machine and GDASApp can we do:
`machine=${machine:-orion}` and `GDASApp=${GDASApp:-/path/to/stuff}`? that way an external script can call this directly without needing to edit it? (thinking for CI)

File: ush/ufoeval/test_yamls.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+# test_yamls.sh 
+# This scripts runs run_ufo_hofx_test.sh for the input YAMLs in ../../parm/atm/obs/testing
+# and outputs pass/fail information
+
+# Work directory
+WORKDIR=$STMP/test_yamls
```
Comment (CoryMartin-NOAA) [inline ush/ufoeval/test_yamls.sh]: what if the user does not have `$STMP` defined?

File: ush/ufoeval/run_ufo_hofx_test.sh [hunk 5]
```diff
@@ -133,17 +170,32 @@ cd $workdir
...
    echo "*************** Running UFO failed ****************"
-   echo "Running EVA for diagnostic purposes"
+   RC=1
+   if [ $run_eva == YES ]; then
+       echo "Running EVA for diagnostic purposes"
+   else
+       exit 1
+   fi
 else
    echo "Running EVA"
```
Comment (CoryMartin-NOAA) [inline ush/ufoeval/run_ufo_hofx_test.sh]: Should this echo here? seems it should only do so if the `if` is met below

**Comment by CoryMartin-NOAA** [comment]
> > +1 for the `boo` \ `yay` output, which is admittedly the only thing I have reviewed in this PR. Are you planning to add the tests to the `ci` eventually?
> 
> I believe @CoryMartin-NOAA is keep to incorporate this into the CI.

Regarding implementing this as testing. What do we all think, should this run for every PR as part of the "workflow testing"? Or should it be a bit more sparse and only run daily on develop? I'm fine either way but suspect it might take a nontrivial amount of time to run if it is a `ctest`.

File: ush/ufoeval/run_ufo_hofx_test.sh (hunk 2 above)
Comment (ADCollard) [reply ush/ufoeval/run_ufo_hofx_test.sh]: @CoryMartin-NOAA I got a little confused here.  But it should just be 
```
if [ $# -ne 1 ]; then
```

The `shift` takes care of the rest.

Unless I misunderstand what you are asking

File: ush/ufoeval/test_yamls.sh (hunk 4 above)
Comment (ADCollard) [reply ush/ufoeval/test_yamls.sh]: The intention was for this line to be edited.  What would you suggest?

File: ush/ufoeval/run_ufo_hofx_test.sh (hunk 2 above)
Comment (CoryMartin-NOAA) [reply ush/ufoeval/run_ufo_hofx_test.sh]: the "or greater than 2" part was what I was concerned about

File: ush/ufoeval/run_ufo_hofx_test.sh (hunk 1 above)
Comment (ADCollard) [reply ush/ufoeval/run_ufo_hofx_test.sh]: This is not correct (there were many iterations!): `[cycle]` removed

File: ush/ufoeval/run_ufo_hofx_test.sh (hunk 2 above)
Comment (CoryMartin-NOAA) [reply ush/ufoeval/run_ufo_hofx_test.sh]: But I see what you're saying, I've never used this like this but it is basically treating the args differently. As long as it works, it's good with me.

File: ush/ufoeval/test_yamls.sh (hunk 4 above)
Comment (CoryMartin-NOAA) [reply ush/ufoeval/test_yamls.sh]: This is fine, just was being thorough with the review.

File: ush/ufoeval/run_ufo_hofx_test.sh (hunk 5 above)
Comment (ADCollard) [reply ush/ufoeval/run_ufo_hofx_test.sh]: Correct.  Removing line.

**Review by CoryMartin-NOAA — APPROVED** [review]
thanks @ADCollard for this and addressing comments!

File: ush/ufoeval/run_ufo_hofx_test.sh (hunk 3 above)
Comment (ADCollard) [reply ush/ufoeval/run_ufo_hofx_test.sh]: I made the changes.  Hope you like them.

**Pull request merged**